CorsixTH/Lua/hospital.lua

```diff
@@ -1762,6 +1765,101 @@ function Hospital:countRoomOfType(type)
   return result
 end
 
+--! Get the number of reception desks in the hospital.
+--!return (int) Number of reception desks in the hospital.
+function Hospital:countReceptionDesks()
+  -- TODO Breaks in multiplayer mode.
+  return self.world.object_counts["reception_desk"]
+end
+
+--! Get the number of benches in the hospital.
+--!return (int) Number of benches in the hospital.
+function Hospital:countBenches()
+  -- TODO Breaks in multiplayer mode.
+  return self.world.object_counts["bench"]
+end
+
+--! Get the number of radiators in the hospital.
+--!return (int) Number of radiators in the hospital.
+function Hospital:countRadiators()
+  -- TODO Breaks in multiplayer mode.
+  return self.world.object_counts["radiator"]
+end
+
+--! Get the number of plants in the hospital.
+--!return (int) Number of plants in the hospital.
+function Hospital:countPlants()
+  -- TODO Breaks in multiplayer mode.
+  return self.world.object_counts["plant"]
+end
+
+--! Get the number of fire extinguishers in the hospital.
+--!return (int) Number of fire extinguishers in the hospital.
+function Hospital:countFireExtinguishers()
+  -- TODO Breaks in multiplayer mode.
+  return self.world.object_counts["extinguisher"]
+end
+
+--! Get the number of general objects in the hospital.
+--!return (int) Number of general objects in the hospital.
+function Hospital:countGeneralObjects()
+  -- TODO Breaks in multiplayer mode.
+  return self.world.object_counts["general"]
+end
+
+--! A new object has been placed in the hospital.
+--!param entity (Entity) The entity that was just placed.
+--!param id (string) That entity's id.
+function Hospital:objectPlaced(entity, id)
+  -- If it is a bench we're placing, notify queueing patients in the vicinity
+  if id == "bench" then
+    local notify_distance = 6
+    local w, h = self.world.map.th:size()
+    for tx = math.max(1, entity.tile_x - notify_distance), math.min(w, entity.tile_x + notify_distance) do
+      for ty = math.max(1, entity.tile_y - notify_distance), math.min(h, entity.tile_y + notify_distance) do
+        for _, patient in ipairs(self.world.entity_map:getHumanoidsAtCoordinate(tx, ty)) do
+          if class.is(patient, Patient) then
```

Comment:
At least it doesn't search through all patients :) 

It's placement of a bench, so the code isn't run often. Whether a player hospital check is correct, I don't know.

In a multi-player situation, I can see the opponent hospital including, I would guess position of all patients etc.
So if an opponent places a bench, how does my game know which patients are affected? Do I run the code at the 'same time' as the opponent computer, so we end up with the same result, or is it run at the opponent only, and does it send the result to me?

Reply:
The scenario of Player1 placing a bench in Player1 hospital shouldn't affect patients in Player2 hospital. It will affect Player2 world (and Player2s representation of Player1 hospital) of course but shouldn't cause Player2s patients to run onChangeQueuePosition. Again its really getting ahead of ourselves on this, just you had the TODO on the counts.

Reply:
Ah right, I didn't even notice that :) 

What I meant is that obviously, moving patients to new benches in player 2 hospital should be viewable at the player 1 computer. Thus somehow, it must know which patient is moving. There are 2 options for that. Either player 1 computes it by itself (ie duplicating the execution of player 2, and giving the same result), or player 2 computes it and sends the result to player 1.
The former costs less network bandwidth, but the code must be executed by all players, the latter is simpler at the cost of additional network traffic.

Depending on what you pick here, bench placement code may also need to be executed by every hospital.